.github/workflows/build-images-workflow-run.yml

```diff
@@ -1,561 +0,0 @@
-      runsOn: ${{ github.repository == 'apache/airflow' && '["self-hosted"]' || '["ubuntu-20.04"]' }}
-    steps:
-      - name: "Checkout ${{ github.ref }} ( ${{ github.sha }} )"
-        uses: actions/checkout@v2
-        with:
-          persist-credentials: false
-          submodules: recursive
-      - name: "Get information about the original trigger of the run"
-        uses: ./.github/actions/get-workflow-origin
-        id: source-run-info
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          sourceRunId: ${{ github.event.workflow_run.id }}
-      - name: "Cancel duplicated 'CI Build' runs"
-        uses: ./.github/actions/cancel-workflow-runs
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          cancelMode: allDuplicates
-          sourceRunId: ${{ github.event.workflow_run.id }}
-      - name: "Output BUILD_IMAGES"
-        id: build-images
-        run: |
-          # Workaround - jobs cannot access env variable in "ifs"
-          # https://github.community/t/how-to-set-and-access-a-workflow-variable/17335/16
-          echo "::set-output name=buildImages::${BUILD_IMAGES}"
-      - name: "Cancel duplicated 'Build Image' runs"
-        # We find duplicates of our own "Build Image" runs - due to a missing feature
-        # in GitHub Actions, we have to use Job names to match Event/Repo/Branch matching
-        # trick ¯\_(ツ)_/¯. We name the build-info job appropriately
-        # and then we try to find and cancel all the jobs with the same Event + Repo + Branch as the
-        # current Event/Repo/Branch combination.
-        uses: ./.github/actions/cancel-workflow-runs
-        with:
-          cancelMode: namedJobs
-          token: ${{ secrets.GITHUB_TOKEN }}
-          notifyPRCancel: true
-          jobNameRegexps: >
-            [".*Event: ${{ steps.source-run-info.outputs.sourceEvent }}
-            Repo: ${{ steps.source-run-info.outputs.sourceHeadRepo }}
-            Branch: ${{ steps.source-run-info.outputs.sourceHeadBranch }}.*"]
-        if: env.BUILD_IMAGES == 'true'
-      - name: "Cancel all 'CI Build' runs where some jobs failed"
-        # We find any of the "CI Build" workflow runs, where any of the important jobs
-        # failed. The important jobs are selected by the regexp array below.
-        # We also produce list of canceled "CI Build' runs as output, so that we
-        # can cancel all the matching "Build Images" workflow runs in the two following steps.
-        # Yeah. Adding to the complexity ¯\_(ツ)_/¯.
-        uses: ./.github/actions/cancel-workflow-runs
-        id: cancel-failed
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          cancelMode: failedJobs
-          sourceRunId: ${{ github.event.workflow_run.id }}
-          notifyPRCancel: true
-          jobNameRegexps: >
-            ["^Pylint$", "^Static checks", "^Build docs$", "^Spell check docs$",
-             "^Provider packages", "^Checks: Helm tests$", "^Test OpenAPI*"]
```

Comment:
Oh, this step is still probably needed, meaning we can't remove the cancel-workflow_runs action entirely.

Reply:
It's an optimisation only, and I am not sure if thats' so much needed after we sped up most of the builds.

Reply:
Selective checks already take care about most savings we have.